### ISSUE TITLE
Incorrect output

### ISSUE THREAD

**Issue by fabien** [alevroub/prettier-plugin-groq#3]
Hi @alevroub,

I think I found a bug with the following projection:

```
{ 'variants': variants[]{ 'units': coalesce(units, 0) }.units }
```

Being incorrectly rendered into:

```
{
  variants[] {
    "units": coalesce(units, 0)
  }.units
}
```

In the case where an array projection is reduced to a single value, the `'variants':` should be preserved. Currently I get `string literal expected` as an error for this.

**Comment by fabien** [comment]
I was able to rewrite it for this particular case into `"variants": coalesce(variants[].units, 0)` which is idempotent. However, I can imagine other scenarios that actually require the intermediate projection as shown above.

**Comment by alejandrovrojas** [comment]
Good catch, that would return the wrong result. The best solution I could come up with is just to avoid merging key/value pairs if the value is accessing an object/projection attribute. [This would be the end result](https://github.com/alevroub/prettier-plugin-groq/blob/master/test/output.groq#L77). Fixed in `0.2.4`. Thanks!
